### PR TITLE
Theme controller and UI model

### PR DESCRIPTION
**Pull request by zbarnz**
Added themes you know you were there. Fixed tests and everything is working as it should. Next step for this branch should be to add more themes and create some kind of modal for theme picking or something

### CODE REVIEW

File: api/lib/controllers/ui.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      const ui = await Ui.findOne({ user_id });
+
+      res.status( 200 ).send( ui );
+    } catch ( err ) {
+      log.error( 'Error fetching ui: ' + err.message );
+    }
+  },
+
+  save: async( req, res ) => {
+    const theme = req.body.theme;
```
Comment (thudsonbu) [inline api/lib/controllers/ui.js:18]: Can you put this inside the try catch block.

File: www/components/DropDown.js [hunk 2]
```diff
@@ -3,17 +3,26 @@ import styles from '../styles/DropDown.module.css';
...
 import { useSelector, useDispatch } from 'react-redux';
+import { useState } from 'react';
 
 const selectUser = ( state ) => state.user;
 
 const DropDown = () => {
   const dispatch = useDispatch();
   const user     = useSelector( selectUser );
   const user_id  = user._id || '';
+  const [ theme, setTheme ] = useState('default');
```
Comment (thudsonbu) [inline www/components/DropDown.js:21]: I don't think we are using this.

File: www/components/DropDown.js [hunk 3]
```diff
@@ -3,17 +3,26 @@ import styles from '../styles/DropDown.module.css';
...
 import AccountCircleIcon from '@material-ui/icons/AccountCircle';
 import LogoutIcon from '@material-ui/icons/ExitToApp';
+import ThemeIcon from '@material-ui/icons/ColorLens';
 
+import { pickTheme } from '../store/features/ui/uiSlice';
+import { changeTheme } from '../utils/theme';
 import { userLogout } from '../store/features/user/userSlice';
 
 import { useSelector, useDispatch } from 'react-redux';
+import { useState } from 'react';
```
Comment (thudsonbu) [inline www/components/DropDown.js:13]: This as well.

File: www/store/features/ui/uiSlice.js [hunk 4]
```diff
@@ -24,3 +32,53 @@ export const toggleDrawer = () => {
...
+
+    try {
+      await client.post(
+        '/ui',
+        { theme, user_id }
+      );
+
+      dispatch({ type: 'ui/pickTheme', payload: { theme } });
+
+      changeTheme( theme );
```
Comment (thudsonbu) [inline www/store/features/ui/uiSlice.js:49]: Not sure we need this. We should be using the store for that.

File: api/lib/controllers/ui.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      const ui = await Ui.findOneAndUpdate(
+        { user_id },
+        { theme, user_id },
+        {
+          upsert: true,
+          new: true
+        }
+      );
+
+      console.log('test');
```
Comment (thudsonbu) [inline api/lib/controllers/ui.js:31]: We should probably cut that out.

**Comment by zbarnz** [comment]
@thudsonbu this seems to have broken a couple things, user refresh no longer seems to be working. Nor are themes. Looking into

**Review by thudsonbu — APPROVED** [review]

**Pull request merged**